Define Heading levels in Settings App

alekhyareddy28 opened the pull request:
## Summary of the Pull Request

_What is this about?_

This PR defines heading levels in the settings app to make scanning more efficient and accessible.

## PR Checklist
* [x] Applies to #7033
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Info on Pull Request

_What does this include?_

- The narrator supports this functionality when a visually impaired user would like to skim through the application by reading only it's headings. They can do so by following these steps - 
  - Enable narrator
  - Press Caps+Space to enable scan mode in the narrator
  - Press H to read the headings
- Before this PR we did not have any headings defined, therefore the narrator would just say no headings defined when H is pressed.
- This PR adds two levels of headings, level1 for the main heading and level2 for the sub-headings within the settings app.

## Validation Steps Performed

_How does someone test & validate?_

Follow these steps to validate that headings are defined - 
  - Enable narrator
  - Press Caps+Space to enable scan mode in the narrator
  - Press H to read the headings

ryanbodrug-microsoft reviewed (approved):
LGTM!

alekhyareddy28 commented:
The pipeline for this PR has passed - https://dev.azure.com/ms/PowerToys/_build/results?buildId=117907&view=results. @crutkas, @ryanbodrug-microsoft it is ready to merge.

niels9001 reviewed (commented):
@alekhyareddy28 Question: is it also possible to set the HeadingLevel in the style of the TextBlock? This would make it future proof since we use the SettingsGroupStyle.

alekhyareddy28 commented:
> @alekhyareddy28 Question: is it also possible to set the HeadingLevel in the style of the TextBlock? This would make it future proof since we use the SettingsGroupStyle.

Thanks @niels9001 that's a great suggestion!
I updated the PR to set the heading level in the style to avoid duplication as you suggested.
The general page, run as admin was however using the default style as we probably don't want a huge margin so I added the heading level2 only for that control. The rest of them are using the heading level set in the GroupStyle.

crutkas commented:
https://dev.azure.com/ms/PowerToys/_build/results?buildId=117958&view=results is the CI for this

crutkas commented:
this succeeded BTW

niels9001 reviewed (approved):
LGTM

ryanbodrug-microsoft commented:
This was successfully built here: https://dev.azure.com/ms/PowerToys/_build/results?buildId=117958&view=results
